### PR TITLE
vecs-school-v1

### PR DESCRIPTION
**Pull request by smunoz1988**
team: @christianonoh , @vvoo21, @Ellyboi , @smunoz1988 
### Description:

This pull request includes changes and additions that fulfill the general, React, Ruby, and project requirements outlined for the vecs-school application. The code changes ensure lint-free code, follow proper GitFlow practices, incorporate React and Ruby best practices, and adhere to the provided design guidelines.

### Changes Made:

1. **Linter Errors:** Reviewed and resolved all linter errors to ensure code cleanliness and consistency.
2. **GitFlow:** Followed the GitFlow branching model, creating feature branches and merging them into the appropriate branches.
3. **Documentation:** Added comprehensive documentation to the codebase, explaining key design decisions, usage instructions, and component explanations.
4. **Ruby and React Best Practices:** Incorporated recommended Ruby best practices such as proper variable naming, modular code structure, and adherence to the DRY (Don't Repeat Yourself) principle.
5. **Design Guidelines:** Implemented the design guidelines provided, ensuring consistency in colors, typography, layout, and interactions.
6. **Original Design Credit:** Provided proper attribution to the original design by Murat Korkmaz in the README, as required by the Creative Commons license.

### Added Features:

1. **Sign-Up and Log-In Pages:** Designed user-friendly sign-up and log-in pages and implemented form validation for mandatory fields.
2. **Home Page (Course Page):** Developed the course page where users can view course information and implemented navigation to the different pages trow the navigation bar.
4. **Details Page:** Show details for the selected course.
5. **Reservations Page:** Show a list of the reservations from the current user.
6. **"Add a New Course Page:** Designed a form for users to create new courses. Implemented validation for mandatory fields. Users can save and navigate back to the home page.
7. **"Add a New Reservation Page:** Designed a form for users to create new reservations.
8. **"Delete Course Page:** Show a list of the courses and let the user delete them

### Technical Requirements:
** Used Postgres as the database and incorporated the Devise gem for authentication. Validated user input to prevent malicious activities and used an appropriate view template engine.

### Deployment:
**Deployed Online:** The project has been successfully deployed and is accessible online, fulfilling the project requirements.### **Database Schema**:
The database schema has been structured according to the provided guidelines.

### Additional Features:

**Sign-Up and Log-In Pages:** Designed user-friendly sign-up and log-in pages and implemented form validation for mandatory fields.

**Please review the changes in this pull request and provide feedback. Thank you!**

### CODE REVIEW

File: src/redux/slices/reservationsSlice.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+//             headers: {
+//                 'Content-Type': 'application/json',
+//                 'Authorization': `Bearer ${token}`,
+//             },
+//         });
+//     }
+//     catch(error){
+//         console.log(error);
+//     }
+// });
```
Comment (Meltrust) [inline src/redux/slices/reservationsSlice.js]: - [ ] Kindly, remove this old commented code in order to improve the repo's cleanliness.  👍

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import Signup from './routes/Signup';
+import Courses from './routes/Courses';
+import Details from './routes/Details';
+import Reserve from './routes/Reserve';
+import Reservations from './routes/Reservations';
+import AddCourse from './routes/AddCourse';
+import DeleteCourse from './routes/DeleteCourse';
+
+
+function App() {
```
Comment (Meltrust) [inline src/App.jsx]: - [ ]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry (I found 3):

image


That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @smunoz1988,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend, both mobile and desktop✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/redux/slices/reservationsSlice.js (hunk 1 above)
Comment (smunoz1988) [reply src/redux/slices/reservationsSlice.js]: Thanks @Meltrust for the review. Changes done! our code is cleanliness!

File: src/App.jsx (hunk 2 above)
Comment (smunoz1988) [reply src/App.jsx]: @Meltrust Thanks for the suggestion! 

We refactor our code and all functions are arrow functions.

**Review by Meltrust — APPROVED** [review]
#### Hi @smunoz1988,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Functions are now ES6✔️ 
- Commented code has been removed✔️ 
- Superb job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**